pandas/_libs/interval.pyx

```diff
@@ -423,7 +423,14 @@ cdef class Interval(IntervalMixin):
             msg = '`other` must be an Interval, got {other}'
             raise TypeError(msg.format(other=type(other).__name__))
 
+        def is_empty(intv):
+            return intv.left==intv.right and intv.closed != 'both'
+
+        if is_empty(self) or is_empty(other):
```

Comment:
Like you commented, I wouldn't be opposed to adding an `is_empty` method .  Could be a little confusing with `IntervalIndex.empty` existing, but I'm fine with that.

We can save this for a future PR though.  In the meantime, can you add a `TODO` comment indicating that we should use the `is_empty` property when it's implemented?  Same deal in the `IntervalArray` code.

Reply:
Done.

Reply:
btw, I started by adding it but though better. Really confusing about how to add one. should it be a property like `closed_right` or a function like `isna`? properties are usually cheap lookups, but in `IntervalArray` the vectorized property calculates a whole array. Should it be `empty`, `isempty` or `is_empty` there's an argument for each.

Like you said. in the future. Having an `empty` method and using it liberally is also a kind of standin for having an `empty` singleton, as discussed, which has no reference to a specific point and is identical+equal for "different" empty intervals. So maybe that's required, except that you'd need to add an explicit empty check in some places, everywhere. in `__eq__` for example.